packages/adapters/solana/src/providers/AuthProvider.ts

```diff
@@ -170,7 +170,7 @@ export class AuthProvider extends ProviderEventEmitter implements SolanaProvider
 
   // -- Private ------------------------------------------- //
   private serializeTransaction(transaction: AnyTransaction) {
-    return base58.encode(transaction.serialize({ verifySignatures: false }))
+    return base58.encode(transaction.serialize({ verifySignatures: false }) as Uint8Array)
```

Comment:
We shouldn't require to do type casting unless there is a specific reason. Why we needed to do this? Is there a version mismatch?

Reply:
We'd better to expilicitly comment this requirement if we cannot remove type casting but this is a bit annoying

Reply:
can try remove this but there is a weird typing that is basically Buffer & Uint8Array that was messing everything up